disnake/ext/commands/bot_base.py

```diff
@@ -433,18 +433,16 @@ def help_command(self) -> Optional[HelpCommand]:
 
     @help_command.setter
     def help_command(self, value: Optional[HelpCommand]) -> None:
+        if value is not None and not isinstance(value, HelpCommand):
```

Comment:
why is this changed?

Reply:
Cleaning up some slightly convoluted logic, which was also partially the cause for a pyright bug with inferring the attribute type correctly; the runtime behavior should be unchanged.